src/consumer.rs

```diff
@@ -408,6 +408,38 @@ pub async fn from_substrait_rex(e: &Expression, input_schema: &DFSchema, extensi
                 "unsupported field ref type".to_string(),
             )),
         },
+        Some(RexType::IfThen(if_then)) => {
+            // Parse `ifs`
+            // If the first element does not have a `then` part, then we can assume it's a base expression
+            let mut when_then_expr: Vec<(Box<Expr>, Box<Expr>)> = vec![];
+            let mut expr = None;
+            for (i, if_expr) in if_then.ifs.iter().enumerate() {
+                if i == 0 {
+                    // Check if the first element is type base expression
+                    match if_expr.then {
+                        Some(_) => (),
+                        None => {
```

Comment:
This might be simpler implemented something like `if if_expr.than.is_none()` rather than using a `match`